### PR TITLE
feat: add sonarqube conventions

### PR DESCRIPTION
**Pull request by yahorbarkouski**

### CODE REVIEW

File: sonarqube-conventions/src/main/kotlin/com/tailrocks/gradle/SonarqubePlugin.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+class SonarqubePlugin : Plugin<Project> {
+
+    override fun apply(project: Project) {
+        project.plugins.apply(org.sonarqube.gradle.SonarQubePlugin::class.java)
+
+        val model = project.extensions.getByType(org.sonarqube.gradle.SonarQubeExtension::class.java)
+        model.properties {
+            property("sonar.sourceEncoding", "UTF-8")
+            property("sonar.verbose", true)
+            property("sonar.host.url", System.getenv("SONAR_HOST"))
```
Comment (donbeave) [inline sonarqube-conventions/src/main/kotlin/com/tailrocks/gradle/SonarqubePlugin.kt:30]: Let's rename to SONAR_HOST_URL

File: sonarqube-conventions/src/main/kotlin/com/tailrocks/gradle/SonarqubePlugin.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            property("sonar.login", System.getenv("SONAR_TOKEN"))
```
Comment (donbeave) [inline sonarqube-conventions/src/main/kotlin/com/tailrocks/gradle/SonarqubePlugin.kt:31]: Rename to SONAR_LOGIN

File: sonarqube-conventions/src/main/kotlin/com/tailrocks/gradle/SonarqubePlugin.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            property("sonar.projectVersion", System.getenv("GIT_COMMIT_SHORTHASH") ?: "")
```
Comment (donbeave) [inline sonarqube-conventions/src/main/kotlin/com/tailrocks/gradle/SonarqubePlugin.kt:32]: Rename to SONAR_PROJECT_VERSION

File: sonarqube-conventions/src/main/kotlin/com/tailrocks/gradle/SonarqubePlugin.kt [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            if (System.getenv("CI_COMMIT_REF_NAME") != "master") {
+                property("sonar.pullrequest.key", System.getenv("CI_EXTERNAL_PULL_REQUEST_IID"))
+                property("sonar.pullrequest.branch", System.getenv("CI_EXTERNAL_PULL_REQUEST_SOURCE_BRANCH_NAME"))
+                property("sonar.pullrequest.base", System.getenv("CI_EXTERNAL_PULL_REQUEST_TARGET_BRANCH_NAME"))
+                property("sonar.scm.revision", System.getenv("CI_EXTERNAL_PULL_REQUEST_SOURCE_BRANCH_SHA"))
+                property("sonar.pullrequest.provider", System.getProperty("sonar.pullrequest.provider", "github"))
```
Comment (donbeave) [inline sonarqube-conventions/src/main/kotlin/com/tailrocks/gradle/SonarqubePlugin.kt:38]: Is it default value `github`?

File: sonarqube-conventions/src/main/kotlin/com/tailrocks/gradle/SonarqubePlugin.kt [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            } else {
+                property("sonar.branch.name", System.getenv("CI_COMMIT_REF_NAME") ?: "master")
```
Comment (donbeave) [inline sonarqube-conventions/src/main/kotlin/com/tailrocks/gradle/SonarqubePlugin.kt:40]: Let's not autofill with master, if env is empty, just put null or don't add this property.

**Review by donbeave — CHANGES_REQUESTED** [review]

File: sonarqube-conventions/src/main/kotlin/com/tailrocks/gradle/SonarqubePlugin.kt (hunk 4 above)
Comment (yahorbarkouski) [reply sonarqube-conventions/src/main/kotlin/com/tailrocks/gradle/SonarqubePlugin.kt:41]: yeah

**Review by donbeave — CHANGES_REQUESTED** [review]

**Review by donbeave — APPROVED** [review]

**Pull request merged**